src/native/fs/fs_napi.cpp

```diff
@@ -1387,17 +1429,17 @@ void
 fs_napi(Napi::Env env, Napi::Object exports)
 {
     auto exports_fs = Napi::Object::New(env);
+    uv_lib_t *lib = (uv_lib_t*) malloc(sizeof(uv_lib_t));
 
-    const char* env_p = std::getenv("GPFS_DL_PATH");
-    if (env_p != NULL) {
+    if(const char* env_p = std::getenv("GPFS_DL_PATH")) {
         LOG("FS::GPFS GPFS_DL_PATH=" << env_p);
-        uv_lib_t *lib = (uv_lib_t*) malloc(sizeof(uv_lib_t));
         if (uv_dlopen(env_p, lib)) {
             PANIC("Error: %s\n" << uv_dlerror(lib));
         }
-        if (uv_dlsym(lib, "gpfs_fcntl", (void **) &dlsym_gpfs_fcntl)) {
+        if (uv_dlsym(lib, "gpfs_linkat", (void **) &gpfs_linkat)) {
             PANIC("Error: %s\n" << uv_dlerror(lib));
         }
+        exports_fs["temp_file"] = Napi::Function::New(env, api<Linkat>);
```

Comment:
Something is off here - when GPFS_DL_PATH we export the generic linkat action as a temp_file symbol ??
What's going on? Can we organize this a bit?

So the the Linkat struct is unrelated specifically to gpfs, it just uses the posix linkat function, and this is unused right now from the js code, because we don't have a use case for it.

The linkfileat function which is only implemented for gpfs is added to any file wrap.

I suggest:
1. Set the generic linkat function as exports_fs["linkat"].
2. Set a generic linkfileat function as a file wrap function, which will currently be implemented only for gpfs and panic otherwise.
3. The js code can check for the the existence of the `GPFS_DL_PATH` env and `fs_account_config.backend === 'GPFS'` to decide if it should use the tmpfile path.

Reply:
maybe in another PR ?

Reply:
I just mean to make a simple fix actually to just (in this scope we simply want to mark that we have loaded the library and that's it, nothing to do with the Linkat api, and nothing to do with temp_file).

```suggestion
        auto gpfs = Napi::Object::New(env);
        // for now we export an (empty) object, which can be checked to indicate that 
        // gpfs lib was loaded and its api's can be used.
        // e.g: gpfs["version"] = Napi::String::New(env,  gpfs_get_version());
        // e.g: gpfs["foo"] = Napi::Function::New(env, api<Foo>);
        gpfs.Freeze();
        exports_fs["gpfs"] = gpfs;
```

Reply:
done

Reply:
Ah right, I see the problem, please add this define here in [src/native/util/napi.h](https://github.com/noobaa/noobaa-core/blob/0a1c7746ad68621fa47b53fb53ecf6b7f9016442/src/native/util/napi.h#L5) right before including the node_api.h and this should include the support for Freeze():
```cpp
#define NAPI_VERSION 8
#include <node_api.h>
```

Reply:
Every thing will work without the gpfs.Freeze(); should I delete it for now ?

Reply:
ok never mind, just comment that Freeze() call and I'll resolve it later.

Reply:
yeah ok, just delete the call to Freeze() and keep the rest.
I'll take a look at how to enable the native freeze api in another PR.

Reply:
ah I see you did that already, so that's done. resolving.